### ISSUE TITLE
Parallelized GALE

### ISSUE THREAD

**Issue by bigfatnoob** [rahlk/Experimental-Algorithms#6]

**Comment by rahlk** [comment]
@bigfatnoob Can you describe you're parallelization strategy, also include other strategies that you think might work.

**Comment by bigfatnoob** [comment]
Yeah will do that

**Comment by rahlk** [comment]
Reply to this thread. Also, is this your strategy — http://charm.cs.uiuc.edu/research/masterSlave?

**Comment by rahlk** [comment]
Or, this? 

> **Island Model:** In the "island" approach to parallelization of genetic programming, the population for a given run is divided into semi-isolated subpopulations (called demes). Each subpopulation is assigned to a separate processor of the parallel computing system. The run begins with the one-time random creation of a separate population of individuals at each processor of the parallel computer system. A typical run of genetic programming on today's computers might entail randomly creating between perhaps 500 and 30,000 individuals at each processor. This process of initial random creation takes place in parallel at each processor. As soon as each separate processor finishes this one-time task, it begins the main generational loop.

**Comment by bigfatnoob** [comment]
Yeah this perfectly describes what I have done. Coincidence. I havent read this. Will include it

**Comment by rahlk** [comment]
Island Model? Or MasterSlave?

**Comment by bigfatnoob** [comment]
No its not master slave as of now. I will try out master slave as it requires more changes in the implementation

**Comment by rahlk** [comment]
Sure.. I'm making changes offline, you can just paste the parts here. I'll incorporate it.

**Comment by bigfatnoob** [comment]
Cool. Thanks man
